File: Lista/quest07/associacao.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        self.escolas.append(escola)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)
        professor.adicionar_escola(self)

if __name__ == "__main__":
    escola1 = Escola("Escola A")
    escola2 = Escola("Escola B")

    professor1 = Professor("Professor X")
    professor2 = Professor("Professor Y")

    escola1.adicionar_professor(professor1)
    escola1.adicionar_professor(professor2)
    escola2.adicionar_professor(professor1)

    print(f"Professores da {escola1.nome}: ", end="")
    for p in escola1.professores:
        print(p.nome, end=" ")
    print()

    print(f"Escolas do {professor1.nome}: ", end="")
    for e in professor1.escolas:
        print(e.nome, end=" ")

File: Lista/quest07/associacao.java
import java.util.ArrayList;
import java.util.List;

class Professor {
    private String nome;
    private List<Escola> escolas;

    public Professor(String nome) {
        this.nome = nome;
        this.escolas = new ArrayList<>();
    }

    public String getNome() {
        return nome;
    }

    public List<Escola> getEscolas() {
        return escolas;
    }

    public void adicionarEscola(Escola escola) {
        this.escolas.add(escola);
    }
}

class Escola {
    private String nome;
    private List<Professor> professores;

    public Escola(String nome) {
        this.nome = nome;
        this.professores = new ArrayList<>();
    }

    public String getNome() {
        return nome;
    }

    public List<Professor> getProfessores() {
        return professores;
    }

    public void adicionarProfessor(Professor professor) {
        this.professores.add(professor);
        professor.adicionarEscola(this);
    }
}

public class Main {
    public static void main(String[] args) {
        Escola escola1 = new Escola("Escola A");
        Escola escola2 = new Escola("Escola B");

        Professor professor1 = new Professor("Professor X");
        Professor professor2 = new Professor("Professor Y");

        escola1.adicionarProfessor(professor1);
        escola1.adicionarProfessor(professor2);
        escola2.adicionarProfessor(professor1);

        System.out.print("Professores da " + escola1.getNome() + ": ");
        for (Professor p : escola1.getProfessores()) {
            System.out.print(p.getNome() + " ");
        }
        System.out.println();

        System.out.print("Escolas do " + professor1.getNome() + ": ");
        for (Escola e : professor1.getEscolas()) {
            System.out.print(e.getNome() + " ");
        }
    }
}
